tutorials/EmbeddingObjectives.qmd

```diff
@@ -0,0 +1,189 @@
+```
+
+We can check that these are the correct at least numerically by calling
+the [`check_gradient`](@ref)
+
+```{julia}
+check_gradient(M, f, grad_f; plot=true)
+```
+
+and the [`check_Hessian`](@ref), which requires a bit more tolerance in its linearity check
+
+```{julia}
+check_Hessian(M, f, grad_f, Hess_f; plot=true, throw_error=true, atol=1e-15)
+```
+
+While they look reasonable here and were already derived – for the general case this derivation
+might be more complicated.
+
+Luckily there exist two functions in [`ManifoldDiff.jl`](https://juliamanifolds.github.io/ManifoldDiff.jl/stable/) that are implemented for several
+manifolds from [`Manifolds.jl`](https://github.com/JuliaManifolds/Manifolds.jl), namely [`riemannian_gradient`](https://juliamanifolds.github.io/ManifoldDiff.jl/stable/library/#ManifoldDiff.riemannian_gradient-Tuple{AbstractManifold,%20Any,%20Any})`(M, p, eG)` that converts a Riemannian gradient
+`eG=`$\nabla \tilde f(p)$  into a the Riemannain one $\operatorname{grad} f(p)$
+ and [`riemannian_Hessian`](https://juliamanifolds.github.io/ManifoldDiff.jl/stable/library/#ManifoldDiff.riemannian_Hessian-Tuple{AbstractManifold,%20Any,%20Any,%20Any,%20Any})`(M, p, eG, eH, X)
+ which converts the Euclidean Hessian `eH=`$\nabla^2 \tilde f(p)[X]$  into $\operatorname{Hess} f(p)[X]$,
+ where we also require the Euclidean gradient `eG=`$\nabla \tilde f(p)$.
+
+So we can define
+
+```{julia}
+grad2_f(M,p) = riemannian_gradient(M, p, ∇f(get_embedding(M), p))
+```
+
+where formally we would also have to call `embed(M,p)` before passing `p` to the Euclidean gradient,
+but here (for the Grassmann manifold with Stiefel representation) the embedding function is the identity.
+
+Similarly (with formally also requiring `embed(M, p, X)`) for the Hessian
+
+```{julia}
+Hess2_f(M, p, X) = riemannian_Hessian(M, p, ∇f(get_embedding(M), p), ∇²f(get_embedding(M), p, X), X)
+```
+
+And we can again check these numerically,
+
+```{julia}
+check_gradient(M, f, grad2_f; plot=true)
+```
+
+and for numerical stability we check here with the more stable `grad_f` gradient.
+
+```{julia}
+check_Hessian(M, f, grad_f, Hess2_f; plot=true, throw_error=true, atol=1e-14)
+```
+
+which yields the same result, but we see that the Euclidean conversion might be a bit less stable.
+
+Now if we want to use these in optimization we would require these two functions to call e.g.
+
+```{julia}
+p0 = [1.0 0.0; 0.0 1.0; 0.0 0.0; 0.0 0.0; 0.0 0.0]
+r1 = adaptive_regularization_with_cubics(
+    M,
+    f,
+    grad_f,
+    Hess_f,
+    p0;
+    debug=[:Iteration, :Cost, "\n"],
+    return_objective=true,
+    return_state=true,
+)
+q1 = get_solver_result(r1)
+r1
+```
+
+but if you choose to go for the conversions, then, thinking of the embedding and defining two new functions
+might betedious, there is a shortcut for these, which performs the change internally, when necessary
+by specifying `objective_type=:Euclidean`.
```

Comment:
For now that tutorial was write only, I did not yet read it, it was more to have example code, so before you spent too much time on it (feel free to do so, but it might have quite a few quirks still).

Reply:
(after all, you've put the ready-for-review label on the PR)

Reply:
Yes, I should have removed that again before I entered todays afternoon – sorry for that. I was hoping to give it a read today at 10, but with a few meetings and quite some debugging, I think it is working now, but I did not yet read the text – feel free to read :D but I will only do that after a few more administrative duties now.